### PR TITLE
Get api/topics

### PR DESCRIPTION
**Pull request by JohnDMurphy**
-Create a 404 error status and message for the path being incorrect.
-Return the correct data in the correct format for a request and test the data.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+const request = require('supertest');
+const app = require('../app');
+const testData = require('../db/data/test-data');
+const db = require('../db/connection');
+const seed = require('../db/seeds/seed');
+
+afterAll(() => {
+  if (db.end) db.end();
```
Comment (jadelandeg) [inline __tests__/app.test.js]: No need for the conditional here, just do return db.end()

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    });
+
+    it('Should return a 200 with an array of topic objects', async () => {
+      const { body } = await request(app).get('/api/topics').expect(200);
+
+      const topics = body.topics;
+      // check its an array
+      expect(topics).toBeInstanceOf(Array);
+      // Check the array is not empty
+      expect(topics.length);
```
Comment (jadelandeg) [inline __tests__/app.test.js]: consider checking the length === the correct number here

**Review by jadelandeg — CHANGES_REQUESTED** [review]
looking really good! Just a couple of tiny comments to consider

File: __tests__/app.test.js (hunk 1 above)
Comment (JohnDMurphy) [reply __tests__/app.test.js]: Done

File: __tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  if (db.end) db.end();
+});
+
+beforeEach(() => {
+  return seed(testData);
+});
+
+describe('nc-be-news-app', () => {
+  describe('GET /api/topics', () => {
+    it('Should return a 404 with an incorrect path', async () => {
```
Comment (jadelandeg) [inline __tests__/app.test.js]: Would usually expect to see happy path testing first, followed by error testing

File: __tests__/app.test.js (hunk 2 above)
Comment (JohnDMurphy) [reply __tests__/app.test.js]: And done

**Pull request merged**